arch/x86/x86_64/include/uk/asm/arch.h

```diff
@@ -9,4 +11,89 @@
 #include <uk/asm/apic.h>
 #include <uk/asm/pic.h>
 
+#ifndef __PLAT_CMN_X86_CPU_DEFS_H__
```

Comment:
These include guards are not necessary given that you have `__UK_ARCH_X86_64_H__` above. Saying that, `arch.h` in `arm64` is not guarded, but instead it is implicitly included via `lcpu.h`:
```
#ifndef __UKARCH_LCPU_H__
#error Do not include this header directly
#endif
```
So for the sake of consistency you can probably do the same for this file, as any common code that includes `lcpu.h` will also include `arch.h` in both architectures. The alternatlive route, and probably what will happen in the long term, is to decouple `arch.h` in `arm64` from `lcpu.h` and see what breaks, but I would refrain from doing that at this stage and leave it for later.